supervisor/hardware/disk.py

```diff
@@ -22,13 +24,29 @@ def __init__(self, coresys: CoreSys):
         """Init hardware object."""
         self.coresys = coresys
 
-    def is_system_partition(self, device: Device) -> bool:
-        """Return true if this is a system disk/partition."""
+    def is_used_by_system(self, device: Device) -> bool:
+        """Return true if this is a system partition."""
         if device.subsystem != UdevSubsystem.DISK:
             return False
 
-        if device.attributes.get("ID_FS_LABEL", "").startswith("hassos"):
+        # Root
+        if device.minor == 0:
+            for child in device.children:
+                try:
+                    device = self.sys_hardware.get_by_path(child)
+                except HardwareNotFound:
+                    continue
+                if device.subsystem == UdevSubsystem.DISK and device.attributes.get(
+                    "ID_FS_LABEL", ""
+                ).startswith("hassos"):
+                    return True
```

Comment:
```suggestion
                if device.subsystem == UdevSubsystem.DISK
                    if device.attributes.get("ID_FS_LABEL", "").startswith("hassos"):
                        return True
```
Optional for readability